python/deltalake/writer.py

```diff
@@ -162,18 +162,12 @@ def write_deltalake(
 
     if filesystem is None:
         if table is not None:
-            filesystem = pa_fs.PyFileSystem(
-                DeltaStorageHandler(
-                    table._table.table_uri(),
-                    table._storage_options,
-                    table._table.get_py_storage_backend(),
-                )
-            )
-        else:
-            filesystem = pa_fs.PyFileSystem(
-                DeltaStorageHandler(table_uri, storage_options)
+            storage_options = (table._storage_options or {}).update(
+                storage_options or {}
             )
 
```

Comment:
@roeap Does this seems like an acceptable simplification? This would mean the storage options provided will override any ones that were originally specified in the table.

Reply:
I think this should work fine. Right now we have an additional issue though, when is comes to storage options. Not entirely sure, but in some cases environment variables set may take precedence over configuration ... I remember fixing this in some places, but fore S§ I am not 100% sure..

Reply:
isn't environment variable overriding code configuration expected behavior?

Reply:
Hmm, not sure. My personal expectation would be that my explicit config takes precedence over the environment. Looking at e.g. multiple targets it seems to be easier to manage. IIRC the python file systems behave like that.

Reply:
I currently get:

```python
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Generic { store: "S3", source: CreateMultipartRequest { source: Error { retries: 0, message: "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>SignatureDoesNotMatch</Code><Message>The request signature we calculated does not match the signature you provided
```

With these changes.